src/main/scala/org/deepdive/ddlog/DeepDiveLogCompiler.scala

```diff
@@ -200,19 +198,17 @@ class CompilationState( statements : DeepDiveLog.Program, config : DeepDiveLog.C
 
   // Analyze the dependency between statements and construct a graph.
   def analyzeDependency(statements: List[Statement]) = {
-    val stmtByHeadName = (extractionRuleGroupByHead.toSeq ++ inferenceRuleGroupByHead.toSeq ++
+    val stmtByHeadName = (extractionRuleGroupByHead.toSeq ++
       functionCallRuleGroupByOutput.toSeq).groupBy(_._1).mapValues(_.map(_._2).toList)
 
     // Look at the body of each statement to construct a dependency graph
     statements foreach {
       case f : FunctionCallRule => dependencies += {
-        f -> (( Some(f.input) flatMap (stmtByHeadName get _)).toSet.flatten.flatten) }
+        f -> ((( f.q.bodies.flatten collect { case x: BodyAtom => x.name })
+          flatMap (stmtByHeadName get _)).toSet.flatten.flatten) }
```

Comment:
This seems like a pretty overloaded code to be inline and repeated.  Can we extract it as a function, say `collectUsedRelations(bodies: List[List[Body]]): Set[Statement]`?